source/common/sink_impl.cc

```diff
@@ -0,0 +1,109 @@
+#include "common/sink_impl.h"
+
+#include <array>
+#include <filesystem>
+#include <fstream>
+
+#include "external/envoy/source/common/common/logger.h"
+#include "external/envoy/source/common/common/random_generator.h"
+
+namespace Nighthawk {
+namespace {
+
+absl::Status verifyStringIsGuid(absl::string_view s) {
```

Comment:
Can you help me understand why does this code depend on the format at all? Are we parsing something from the UUID or do we depend on some other property of the identifier?

I would expect the code to just use the execution id as an opaque identifier and at most validate that it is unique in this instance of the sink.

Reply:
The file sink implementation depends on us constraining the format to this, so it doesn't have to worry about weirdness with respect to directory names it generates internally based on this. We can work to remove this constraint in the future, but for now this seems the simplest solution.
Any other sink implementation probably wouldn't be subject to this.

Reply:
That makes sense, can we then adjust the function names and the checks they perform to indicate that the goal is to verify the ID can be a directory name rather than trying to verify it is a Guid? E.g. the function could be called `verifyCanBeUsedAsDirectoryName` or similar and it should only perform a check for the containing characters the are supported by a file system.

Reply:
Done in 6a31b48